src/Reflex/Spider/Internal.hs

```diff
@@ -1170,18 +1165,12 @@ buildDynamic readV0 v' = do
   return d
 
 unsafeBuildDynamic :: BehaviorM x (PatchTarget p) -> Event x p -> Dyn x p
-unsafeBuildDynamic readV0 v' = Dyn $ unsafeNewIORef x $ UnsafeDyn x
-  where x = (readV0, v')
+unsafeBuildDynamic readV0 v' =
+  Dyn $ unsafePerformIO $ newIORef $ UnsafeDyn (readV0, v')
```

Comment:
If this gets called *many times*, then it might be worth working around GHC 15127 by using `runRW#` manually to unbox the `IORef`.

```haskell
unsafeNewIORef :: a -> IORef a
unsafeNewIORef a = IORef $ STRef ( runRW# $ \s ->
  case newMutVar# a (noDuplicate# s) of
    (# _s', mv #) -> mv )
```

or

```haskell
unsafeDupableNewIORef :: a -> IORef a
unsafeDupableNewIORef a = IORef $ STRef ( runRW# $ \s ->
  case newMutVar# a s of
    (# _s', mv #) -> mv )
```

Reply:
I don't think it does - because unsafeBuildDynamic is not a very frequently used function. Current benchmarks show no significant difference (for any of these changes)